luskan/src/main/java/org/opencadc/luskan/ws/StorageTapService.java

```diff
@@ -0,0 +1,224 @@
+                return new AvailabilityStatus(false, null, null, null, RestAction.STATE_OFFLINE_MSG);
+            }
+            if (RestAction.STATE_READ_ONLY.equals(state)) {
+                return new AvailabilityStatus(false, null, null, null, RestAction.STATE_READ_ONLY_MSG);
+            }
+
+            String tapDsName = DataSourceProviderImpl.getDataSourceName(null, "tapadm");
+            DataSource tapadm = DBUtil.findJNDIDataSource(tapDsName);
+            InitDatabaseTS tsi = new InitDatabaseTS(tapadm, null, "tap_schema");
+            tsi.doInit();
+
+            String uwsDsName = DataSourceProviderImpl.getDataSourceName(null, "uws");
+            DataSource uws = DBUtil.findJNDIDataSource(uwsDsName);
+            InitDatabaseUWS uwsi = new InitDatabaseUWS(uws, null, "uws");
+            uwsi.doInit();
+
+            // create the TAP schema
+            InitLuskanSchemaContent lsc = new InitLuskanSchemaContent(tapadm, null, "tap_schema");
+            lsc.doInit();
+
+            // run a couple of queries
+            CheckResource cr = new CheckDataSource(tapDsName, TAP_TEST);
+            cr.check();
+
+            cr = new CheckDataSource(uwsDsName, UWS_TEST);
+            cr.check();
+
+            LocalAuthority localAuthority = new LocalAuthority();
+            RegistryClient reg = new RegistryClient();
+
+            // TODO - These do not work for intTest unless the testing environment deploys these services.
+//            URI credURI = localAuthority.getServiceURI(Standards.CRED_PROXY_10.toString());
+//            String url = reg.getServiceURL(credURI, Standards.VOSI_AVAILABILITY, AuthMethod.ANON).toExternalForm();
+//            CheckResource checkResource = new CheckWebService(url);
+//            //checkResource.check();
+//
+//            URI usersURI = localAuthority.getServiceURI(Standards.UMS_USERS_01.toString());
+//            url = reg.getServiceURL(usersURI, Standards.VOSI_AVAILABILITY, AuthMethod.ANON).toExternalForm();
+//            checkResource = new CheckWebService(url);
+//            //checkResource.check();
+//
+//            URI groupsURI = localAuthority.getServiceURI(Standards.GMS_SEARCH_01.toString());
+//            url = reg.getServiceURL(groupsURI, Standards.VOSI_AVAILABILITY, AuthMethod.ANON).toExternalForm();
+//            checkResource = new CheckWebService(url);
+//            checkResource.check();
+
+        } catch (CheckException ce) {
+            // tests determined that the resource is not working
+            isGood = false;
+            note = ce.getMessage();
+        } catch (Throwable t) {
+            // the test itself failed
+            log.error("test failed", t);
+            isGood = false;
+            note = "test failed, reason: " + t;
+        }
+        return new AvailabilityStatus(isGood, null, null, null, note);
+    }
+
+    @Override
+    public void setState(String state) {
+        AccessControlContext acContext = AccessController.getContext();
```

Comment:
Do not need to do any auth check here; AvailablityServlet in cadc-vosi does it from config.